src/lib/reasoners/satml_frontend.ml

```diff
@@ -1335,6 +1341,7 @@ module Make (Th : Theory.S) : Sat_solver_sig.S = struct
       end;
       dep
     | (Util.Timeout | I_dont_know _ ) as e -> raise e
+    | Util.Step_limit_reached n -> i_dont_know env (Step_limit n)
```

Comment:
As I explained, this is probably incorrect (i.e. there is no guarantee that `env` is in a valid state), because there is no guarantee that the sat solver be in a consistent state when the `Step_limit` is reached.

Reply:
For instance we could raise `Steps_limit` in the middle of the `Satml.theory_propagate` loop and end up with the queues in a weird state that may or may not be valid.

Reply:
I am adding a wrapper at the Frontend level that blocks the calls to the SAT (sorry it came just after the review request, I forgot to add it)

Reply:
It does not solve the issue where the number of steps is updated after failing once, which cannot be solved by putting the solver in a "limited" state without making it permanent.
I'm starting to think the fatal error was a good choice while there is no deep work on the solver to make sure environments are consistent even when failing in the middle of the execution of an instruction.

Reply:
> It does not solve the issue where the number of steps is updated after failing once, which cannot be solved by putting the solver in a "limited" state without making it permanent.

If the solver is in a "step limit exceeded" state, that shouldn't change if the step limit is updated, so that should be fine, no?